M2/BUILD/mahrud/Makefile

```diff
@@ -0,0 +1,24 @@
+# 
+# Written for Fedora 26
+#
+
+.PHONY: always
+
+rpm-docker-build:
+	[ ! -z `docker images -q m2-rpm` ] || docker build -t m2-rpm .
+	docker run -it \
+	  -v `pwd`/../../..:/M2 \
+	  m2-rpm \
+	  make rpm-build -j4 -C /M2/M2/BUILD/mahrud
+
+rpm-build:
+	mkdir -p build;
+	make -j4 -C ../..;
+	cd build && ../../../configure \
+	  --enable-rpm \
+	  --prefix=/usr \
+	  --enable-download \
+	  --enable-build-libraries="factory pari flint ntl fflas_ffpack memtailor mathic mathicgb";
```

Comment:
I tried on a fresh Fedora 26:
```
[root@973654e8779c dan]# make -j4
...
/usr/bin/install -c -d submodules/memtailor
make: Entering directory '/M2/M2/BUILD/dan/builds.tmp/-master/submodules/memtailor'
+ /M2/M2/submodules/memtailor/configure --prefix=/M2/M2/BUILD/dan/builds.tmp/-master/usr-host PKG_CONFIG_PATH=/M2/M2/BUILD/dan/builds.tmp/-master/usr-host/lib/pkgconfig: GTEST_PATH=/M2/M2/BUILD/dan/builds.tmp/-master/usr-host/include/gtest AR=ar AS=as DLLTOOL=false OBJDUMP=objdump STRIP=strip 'CXXFLAGS=-std=gnu++11 -g3 -O2' 'CFLAGS=-std=gnu11 -g3 -O2' FFLAGS=-O2 FCFLAGS=-O2 TARGET_ARCH= 'CPPFLAGS=-isystem /M2/M2/include -isystem /M2/M2/BUILD/dan/builds.tmp/-master/include -isystem /M2/M2/BUILD/dan/builds.tmp/-master/usr-host/include  -I/M2/M2/submodules/memtailor/src  -I/M2/M2/submodules/mathic/src  -I/M2/M2/submodules/mathicgb/src  -I/M2/M2/submodules/fflas_ffpack/src -isystem /usr/include/libxml2  -DSING_NDEBUG -DOM_NDEBUG     -DNDEBUG' CC=gcc CXX=g++ 'LDFLAGS=-L/M2/M2/BUILD/dan/builds.tmp/-master/usr-host/lib  -L/M2/M2/BUILD/dan/builds.tmp/-master/submodules/memtailor/.libs  -L/M2/M2/BUILD/dan/builds.tmp/-master/submodules/mathic/.libs  -L/M2/M2/BUILD/dan/builds.tmp/-master/submodules/mathicgb/.libs -g3' LOADLIBES= LDLIBS= 'LIBS=-lxml2 -llzma -lz -llapack -lblas  -L/usr/lib/gcc/x86_64-redhat-linux/7 -L/usr/lib/gcc/x86_64-redhat-linux/7/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/usr/lib/gcc/x86_64-redhat-linux/7/../../.. -ldl -lgfortran -lm -lquadmath -lfactory -lflint -lmpfr -lntl -lgmp -lomalloc -lsingular_resources  -lgdbm -lgc  -lreadline -ltinfo -lpthread -ldl ' --build=x86_64-pc-linux-gnu --with-gtest=yes
checking for gcc... gcc
checking whether the C compiler works... no
configure: error: in `/M2/M2/BUILD/dan/builds.tmp/-master/submodules/memtailor':
configure: error: C compiler cannot create executables
See `config.log' for more details
make[1]: *** [GNUmakefile:171: submodules/memtailor/Makefile] Error 77
make[1]: Leaving directory '/M2/M2/BUILD/dan/builds.tmp/-master'
make: *** [Makefile.include:17: all] Error 2
[root@973654e8779c dan]# git branch
* master
  release-1.10
  translate
[root@973654e8779c dan]# uname -a
Linux 973654e8779c 4.11.11-300.fc26.x86_64 #1 SMP Mon Jul 17 16:32:11 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
[root@973654e8779c dan]# make status -C builds.tmp/-master/
make: Entering directory '/M2/M2/BUILD/dan/builds.tmp/-master'
-----------------------------------------------------------------------------
#  Here is the way you ran /M2/M2/configure before:
#
# '/M2/M2/configure'  '--enable-encap' '--enable-download' '--enable-tarfile' '--enable-strip' '--enable-build-libraries=' 'CC=' 'CXX=' 'FC=' 'CPPFLAGS=' 'LDFLAGS=' '--prefix=/usr' '--build=x86_64-pc-linux-gnu' 'build_alias=x86_64-pc-linux-gnu'
#
#  Type
        make reconfigure
#  to run it that way again.
#
-----------------------------------------------------------------------------
make: Leaving directory '/M2/M2/BUILD/dan/builds.tmp/-master'
[root@973654e8779c dan]# git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working tree clean
```
Hmm, if you'd like to give Docker a try, in a folder containing the Dockerfile run `docker build -t m2-rpm ./`
and then:
```
docker run -it -v `pwd`/../../..:/M2 m2-rpm /usr/bin/bash  # this'll open a shell in a fresh container
make -j4 -C /M2/M2/BUILD/dan
```